Lib/Protocols/IdFTPServer.pas

```diff
@@ -7575,4 +7584,108 @@ procedure TIdFTPServerInfo.SetAdditionalFacts(const AValue: TStrings);
   FAdditionalFacts.Assign(AValue);
 end;
 
+{ TIdFTPClientInfo }
+
+constructor TIdFTPClientInfo.Create;
+begin
+  inherited Create;
+  FFacts := TStringList.Create;
+{$IFDEF HAS_TStringList_CaseSensitive}
+  TStringList(FFacts).CaseSensitive := False;
+{$ENDIF}
+end;
+
+destructor TIdFTPClientInfo.Destroy;
+begin
+  FreeAndNil(FFacts);
+  inherited;
+end;
+
+function TIdFTPClientInfo.GetClientName: String;
+begin
+  Result := FFacts.Values['Name'];
+end;
+
+function TIdFTPClientInfo.GetClientVersion: String;
+begin
+  Result := FFacts.Values['Version'];
+end;
+
+function TIdFTPClientInfo.GetCLNTParams: String;
+begin
+  Result := Trim(ClientName + ' '+ClientVersion + ' '+PlatformDescription);
+end;
+
+function TIdFTPClientInfo.GetCSIDParams: String;
+var i : Integer;
+begin
+  Result := '';
+  for i := 0 to FFacts.Count - 1 do
+  begin
+    Result := '; '+FFacts[i];
+  end;
+  IdDelete(Result,1,2);
+end;
+
+function TIdFTPClientInfo.GetPlatformDescription: String;
+begin
+  Result := FFacts.Values['OS'];
+end;
+
+function TIdFTPClientInfo.GetPlatformVersion: String;
+begin
+  Result := FFacts.Values['OSVer'];
+end;
+
+function TIdFTPClientInfo.GetVendor: String;
+begin
+  Result := FFacts.Values['Vendor'];
+end;
+
+procedure TIdFTPClientInfo.SetClientName(const AValue: String);
+begin
+  FFacts.Values['Name'] := AValue;
+end;
+
+procedure TIdFTPClientInfo.SetClientVersion(const AValue: String);
+begin
+  FFacts.Values['Version'] := AValue;
+end;
+
+procedure TIdFTPClientInfo.SetCLNTParams(const AValue: String);
+var LBuf : String;
+begin
+  LBuf := TrimLeft(Avalue);
+  ClientName := Fetch(LBuf);
+  ClientVersion := Fetch(LBuf);
+  PlatformDescription := Fetch(LBuf);

```

Comment:
Since nothing follows the description, there is no need to use `Fetch()` on this line.